### PR TITLE
Recategorize a few pypy bugs, fix some others

### PR DESCRIPTION
**Pull request by da-woods**
"Fixes" a few bugs that are just caused by very small implementation
details, recategorize some others (e.g. those that cimport array
are not expected to ever work)

### CODE REVIEW

File: tests/run/matrix_multiplier.pyx [hunk 1]
```diff
@@ -22,9 +22,11 @@ ExtMatMult("ExtMatMult('ExtMatMult(1) @ ExtMatMult(2)') @ ExtMatMult(2)")
 >>> x @ y
 Traceback (most recent call last):
 TypeError: unsupported operand type(s) for @: 'int' and 'int'
->>> x @= y
+
+PyPy exception message has '@' rather than '@='
+>>> x @= y  # doctest: +ELLIPSIS
 Traceback (most recent call last):
-TypeError: unsupported operand type(s) for @=: 'int' and 'int'
+TypeError: unsupported operand type(s) for @...: 'int' and 'int'
```
Comment (da-woods) [inline tests/run/matrix_multiplier.pyx:29]: @mattip Not a big issue, but arguable somewhere that PyPy's error reporting could be improved

File: tests/run/test_genericclass.py [hunk 2]
```diff
@@ -162,7 +162,8 @@ class TestClassGetitem(unittest.TestCase):
     # BEGIN - Additional tests from cython
     def test_no_class_getitem(self):
         class C: ...
-        with self.assertRaises(TypeError):
+        # note that PyPy raises AttributeError on __class_getitem__
+        with self.assertRaises(TypeError if not hasattr(sys, "pypy_version_info") else AttributeError):
```
Comment (da-woods) [inline tests/run/test_genericclass.py:166]: @mattip I've worked around it, but I'd think `TypeError` is the right error to raise

File: tests/run/unicodemethods.pyx [hunk 3]
```diff
@@ -156,23 +156,23 @@ def splitlines_keep_bint(unicode s):
...
     ab jd
     <BLANKLINE>
     sdflk as sa
     <BLANKLINE>
-    sadas asdas fsdf 
+    sadas asdas fsdf
     --
     ab jd
     sdflk as sa
-    sadas asdas fsdf
```
Comment (da-woods) [inline tests/run/unicodemethods.pyx:175]: It looks like my editor "helpfully" corrects the trailing whitespace in this file. I'll undo it

File: tests/run/matrix_multiplier.pyx (hunk 1 above)
Comment (mattip) [reply tests/run/matrix_multiplier.pyx:29]: Thanks. I opened https://foss.heptapod.net/pypy/pypy/-/issues/3605

File: tests/run/test_genericclass.py (hunk 2 above)
Comment (mattip) [reply tests/run/test_genericclass.py:166]: This is fixed on PyPy HEAD, thanks

File: tests/pypy_bugs.txt [hunk 4]
```diff
@@ -2,17 +2,15 @@
...
 bufaccess
 memoryview.memoryview$
-sequential_parallel
 
-yield_from_pep380
-memoryview_inplace_division
-
-run.unicodemethods
+# looks like a doctest issue? It's failing to match output that looks
+# identical to by eye
```
Comment (scoder) [inline tests/pypy_bugs.txt:9]:
```suggestion
# identical to my eye
```

**Comment by scoder** [comment]
Thanks, also for the comments. It's always helpful to keep a note of why a test was disabled.

**Pull request merged**